cve_bin_tool/helper_script.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+import re
+
+from cve_bin_tool.cvedb import CVEDB, DBNAME, DISK_LOCATION_DEFAULT
+from cve_bin_tool.extractor import Extractor
+from cve_bin_tool.log import LOGGER
+from cve_bin_tool.strings import Strings
+from cve_bin_tool.util import DirWalk, inpath
+from cve_bin_tool.version_scanner import (  # using it just for clean_file_path
+    VersionScanner,
+)
+
+
+class HelperScript:
+    """Helps Contributors to understand how various common filenames, version-strings, etc. look"""
+
+    dbname = os.path.join(DISK_LOCATION_DEFAULT, DBNAME)
+
+    def __init__(self, filename, vendor_name=None, product_name=None):
+        self.filename = filename
+        self.extractor = Extractor()
+        self.logger = LOGGER.getChild(self.__class__.__name__)
+        self.product = product_name or self.parse_filename(filename)
+        self.vendor = vendor_name
+
+        # for setting the data
+        self.connection = None
+        self.dbpath = os.path.join(DISK_LOCATION_DEFAULT, DBNAME)
+
+        # for extraction
+        self.walker = DirWalk().walk
+        self.file_stack = []
+
+        # for output
+        self.contain_patterns = []
+        self.filename_pattern = []
+        self.version_pattern = []
+        self.vendor_product = []
+
+    def extract_file(self):
+        with self.extractor as ectx:
+            if ectx.can_extract(self.filename):
+                binary_string_list = []
+                for filepath in self.walker([ectx.extract(self.filename)]):
+                    clean_path = VersionScanner.clean_file_path(
+                        filepath
+                    )  # to be removed???
+                    self.file_stack.append(f"{clean_path}")
+                    self.logger.info(f"scanning binaries from {clean_path}")
+                    # go to that path and scan for strings
+                    matches = self.search_pattern(
+                        self.parse_strings(filepath)
+                    )  # list of strings in binary
+                    binary_string_list.append(matches)
+                return binary_string_list
+
+    def parse_strings(self, filepath):
+        s = Strings(filepath)
+        lines = s.parse()
+        return lines
+
+    def search_pattern(self, string_list):
+        pattern = r"" + self.product + r"(\d+.\d+.\d+)?"
```

Comment:
this the raw string pattern that I decided to go with. I would be iterating over it as I test this on more files. Is this implementation okay??

Reply:
Did you mean to use unescaped periods here?  . means "match any one character" and `\.` means "match only the character `.`"   I'm guessing you either want to use either `\.` (match a period) or `[.-]` (match a period or a dash) here.

Reply:
But other than that comment, yes this is a fine place to start iterating.  

f you want an immediate iteration, I'd add a couple of whitespace characters in to the pattern so that you could match, say
product1.2.3
product 1.2.3
product-1.2.3
product.1.2.3

and maybe others.  so that would change your pattern to include a [ -.]? block between the product name and the semantic version pattern.

You might also want to match a 1.2 pattern (i.e. a shorter semantic version) by having some question marks (match 0 or 1 characters) for the last period and a * (match 0 or more) on the last digit.  

And this one is definitely for future iteration, but you'll likely want to handle things like 2.3.4rc1, 2.1.post1, and some of the distro-specific version additions eventually.

Reply:
@ terriko I've updated the regex pattern to `product(.*)([0-9]+[.-][0-9]+([.-][0-9]+)?)`.
The `(.*)` would help to match patterns like
product version 1.2.3
product v1.2.3(1)

You could see if these regex's match at [pythex.org](https://pythex.org/). (I've used this to make regexes for the patterns).